`azurerm_storage_table` - add attribute `resource_manager_id` (#28809)
hashicorp/terraform-provider-azurerm: website/docs/r/storage_table.html.markdown

* `id` - The ID of the Table within the Storage Account.

> * `resource_manager_id` - The ID of the Table in the Resource Manager format. Useful for cases like IAM that do not use the Data Plane ID.

Reviewer: We can match the description with how we've done it in other resources. I'd also caution against specifying how to use this as the docs can get very clogged up if we provided helpful tips on most things

```suggestion
* `resource_manager_id` - The Resource Manager ID of this Storage Table.
```

> * `resource_manager_id` - The ID of the Table in the Resource Manager format, e.g. "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myRG/providers/Microsoft.Storage/storageAccounts/myAccount/tableServices/default/tables/myTable". Useful for cases like IAM that do not use the Data Plane ID.

Author: I was conflicted about mentioning a use-case, whether we should let the user decide on how they might use a feature.
